### DISCUSSION TITLE
Navigation along Cycle Ways

### DISCUSSION THREAD

**Discussion by JulianOestreich90** [osmandapp/OsmAnd#14316] [Maps Related]
I love the OSMAnd App, but there is one thing i'm really missing: More functions for cyclists. When it's about routing i only have the option to specify my driving style ("Fahrweise"). There i can prefer side ways, shorter routes, unpaved roads, but unfortunatly no cylce ways. I added the Cyclosm Online Maps, but i can't make the navigation to route along the cycle ways. Is there something i'm missing or is this functionality simply not supported without intention to change it?

**Comment by JulianOestreich90** [comment]
I used the sideways option, which sometimes improves it a bit. But here you can see how the most direct pink eurovelo route is not preferably chosen and instead i'll be routed the longer way along the river.

![Screenshot_20220517-235053.png](https://user-images.githubusercontent.com/19915127/168916035-b489032d-49cb-478a-a678-c3ecdb33a0af.png)

**Comment by pebogufi** [comment]
I made some tests with higher priorization of cycleroutes (icn, ncn, rcn, lcn) **over everything else**.
Search in the file for "@PB" then "driving_style" to find my changes.
The best result i got with "driving_style_safety", the other two with minor results.
[routing.zip](https://github.com/osmandapp/OsmAnd/files/8732864/routing.zip)

Unzip the file and copy it to subdir "routing"
You have to create a new profile for bike and choose this file.
Be aware: the file is just a experimental example, if you find a better solution, please feel free to post it.

**Reply by fixifoxy** [reply]
when I am going to import Osmand is asking for an OSF formated file.?????

**Reply by JulianOestreich90** [reply]
As pebogufi stated, you need to copy your *.routing.xml into the ..osmand/routing/ folder! Then create a new routing profile inside the app. If you open the routing.xml you will also find detailed instructions inside it.

**Reply by fixifoxy** [reply]
Hi Julian,
see my discussion under #14663 
I am working with the rendering and routing xmls of a french Osmander. 
Published under "Style pour OSMAND.html"
Since I´ve seen the the website I processed the importing of routing and rendering files as   described in the above mentioned
website.
Meanwhile "cricri" the french guy changed to the official published way, but there the OSF format is afforded and in the routing folder no file is shown.
